Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications.py

```diff
@@ -152,6 +253,91 @@ def list_service_principals_command(ms_client: Client, args: dict) -> CommandRes
+
+    Args:
+        args: The arguments were passed with the command.
+
+    Returns:
+        If the two arguments are missing, raise an exception, otherwise return the (object) id.
+        This validation returns tuple, the first element will be sent to the api call, the second is the given id/app id as is.
+    """
+    object_id = args.get('id')
+    app_client_id = args.get('app_id')
+    if not (object_id or app_client_id):
+        raise DemistoException("Either the (object's) `id` or the `application_id` arguments must be provided.")
+
+    # if both are provided, pass the object_id
+    if object_id:
+        return f"/{object_id}", object_id
+    return f"(appId='{app_client_id}')", app_client_id
+
+
+def get_service_principal_command(ms_client: Client, args: dict) -> CommandResults:
+    """
+
+    Args:
+        ms_client: The Client
+        args: demisto.args()
+
+    Returns:
+        Results to post in demisto
+    """
+    id_for_request, service_id = validate_service_principal_input(args=args)
+    results = ms_client.get_single_service_principal(id_for_request)
+    return CommandResults(
+        'MSGraphApplication',
+        'id',
+        outputs=results,
+        readable_output=tableToMarkdown(
+            'Available service (application):',
+            results,
+            headers=['id', 'appId', 'appDisplayName', 'accountEnabled', 'deletedDateTime'],
+            removeNull=True
+        )
+    )
+
+
+def update_service_principal_command(ms_client: Client, args: dict) -> CommandResults:
+    """
+    Update the properties of servicePrincipal object.
+    Args:
+        ms_client: The Client
+        args: demisto.args()
+
+    Returns:
+        an informative message.
+    """
+    id_for_request, service_id = validate_service_principal_input(args=args)
+    fields_to_update = {"add_ins": "addIns", "alternative_names": "alternativeNames",
+                        "app_role_assignment_required": "appRoleAssignmentRequired", "app_roles": "appRoles",
+                        "custom_security_attributes": "customSecurityAttributes", "display_name": "displayName",
+                        "home_page": "homepage", "key_credentials": "keyCredentials", "logout_url": "logoutUrl",
+                        "oauth2_permission_scopes": "oauth2PermissionScopes",
+                        "preferred_single_sign_on_mode": "preferredSingleSignOnMode",
+                        "service_principal_names": "servicePrincipalNames", "tags": "tags",
+                        "token_encryption_key_id": "tokenEncryptionKeyId"}
+    boolean_fields_to_update = {"account_enabled": "accountEnabled", "app_role_assignment_required": "appRoleAssignmentRequired"}
+
+    data = {
+        fields_to_update[field]: args[field]
+        for field in fields_to_update
+        if field in args
+    }
+
+    for field in boolean_fields_to_update:
+        if field in args:
+            data[boolean_fields_to_update[field]] = argToBoolean(args[field])
```

Comment:
shouldn't it be `data[field] = argToBoolean(args[field])`? Or have I misunderstood the intention here

Reply:
boolean_fields_to_update maps between the argument which was passed from the user, to the right key according to the API.
For example the field: "account_enabled", in data we create a new key "accountEnabled" and its value is  args[field] (which means args["account_enabled"] - the input from the user).
So this line is correct